lib/views/profile/widgets/build_edit_location.dart

```diff
@@ -0,0 +1,308 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:animated_custom_dropdown/custom_dropdown.dart';
+
+// firebase
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+// constants
+import 'package:recycle/constants.dart';
+
+// utils
+import 'package:recycle/utils/data_classes.dart';
+
+// widgets
+import 'package:recycle/views/profile/widgets/profile_widget.dart';
+
+class BuildEditLocationField extends ConsumerStatefulWidget {
+  const BuildEditLocationField({
+    super.key,
+    required this.windowWidth,
+    required this.userData,
+    required this.windowHeight,
+  });
+
+  final double windowWidth;
+  final double windowHeight;
+  final Map<String, dynamic> userData;
+
+  @override
+  ConsumerState<BuildEditLocationField> createState() =>
+      _BuildEditLocationFieldState();
+}
+
+class _BuildEditLocationFieldState
+    extends ConsumerState<BuildEditLocationField> {
+  late String userLocation;
+  late String userRegion;
+  late String userPrefecture;
+  late Map<String, List<String>> regionPrefectures;
+
+  @override
+  void initState() {
+    super.initState();
+    userLocation = widget.userData[UserData.keyLocation];
+    userRegion = userLocation.split(',').first.trim();
+    userPrefecture = userLocation.split(',').last.trim();
+    regionPrefectures = {

```

Comment:
Should probably move these to lib/constants.dart

Reply:
those need userData to be called, so putting them in lib/constants would break the code.

Reply:
@SLuo490 Sorry I meant the `regionPrefectures` since its just a map.

Reply:
we can do that, but region prefectures is only used in the edit location widget, whereas the constants are used throughout the application, so I don't want to clutter the constant file if possible.